src/Console/Command/SelfUpdateCommand.php

```diff
@@ -75,21 +90,19 @@ protected function execute(InputInterface $input, OutputInterface $output)
         preg_match('/^v?(?<major>\d+)\./', $currentVersion, $matches);
         $currentMajor = (int) $matches['major'];
 
-        $checker = new NewVersionChecker(new GithubClient());
-
         try {
-            $latestVersion = $checker->getLatestVersion();
-            $latestVersionOfCurrentMajor = $checker->getLatestVersionOfMajor($currentMajor);
+            $latestVersion = $this->versionChecker->getLatestVersion();
+            $latestVersionOfCurrentMajor = $this->versionChecker->getLatestVersionOfMajor($currentMajor);
         } catch (\Exception $exception) {
             $output->writeln(sprintf(
                 '<error>Unable to determine newest version: %s</error>',
                 $exception->getMessage()
             ));
 
-            return 0;
+            return 1;
```

Comment:
good catch :+1: maybe this should go lower versions as well?

Reply:
Target branch is already 2.2 ;)

Reply:
this bugfix deserves dedicated PR that gonna be merged sooner:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3196